src/component/addons/AddonForm/AddonForm.jsx

```diff
@@ -149,7 +149,7 @@ export const AddonForm = ({ editMode, provider, addon, fetch }) => {
                 </a>
                 <p className={commonStyles.error}>{errors.general}</p>
             </section>
-            <form onSubmit={onSubmit}>
```

Comment:
Instead of removing the `onSubmit` handler, maybe adding `type="button"` to the cancel button would work.

(That should stop the cancel button from submitting the form, while still allowing for regular form submissions.)